src/Simulator/FireSimulator.cpp

```diff
@@ -121,29 +142,28 @@ void cFireSimulator::SimulateChunk(std::chrono::milliseconds a_Dt, int a_ChunkX,
 		}
 
 		// Burn out the fire one step by increasing the meta:
+		if (!BurnsForever && (y > 0))
+		{
+			a_Chunk->SetMeta(x, y, z, BlockMeta + 1);
```

Comment:
What if the BlockMeta is already `0x0f`? This will roll over.

Reply:
There isn't anything else changing the BlockMeta, so that wouldn't be an issue. However I've moved this code block back to where it was before anyway.